src/main/java/org/springframework/data/mapping/PropertyPath.java

```diff
@@ -252,6 +281,11 @@ public static PropertyPath from(String source, TypeInformation<?> type) {
 			}
 		}
 
+		if (result == null) {
+			throw new IllegalStateException(
```

Comment:
With `Assert.state(result == null, …)` the IDE doesn't actually consider the variable tested as non-null.